common/src/node/ethereum/provider.rs

```diff
@@ -0,0 +1,22 @@
+use crate::EthError;
+use ethers::providers::{Http, Provider};
+use url::Url;
+
+static DEFIWALLETCORE_AGENTINFO: Option<&'static str> = option_env!("DEFIWALLETCORE_AGENTINFO");
```

Comment:
option_env is inspected at compile time. What if in run time?

Reply:
for runtime, need to generate singleton pattern.
and share with arc X mutex

but agent is hardly changed in runtime, statictime action seems better.

Reply:
if you want runtime, you can possibly use `once_cell` instead of arc+mutex: https://docs.rs/once_cell/1.13.0/once_cell/ so it'd only be set once based on the environment during initialization

Reply:
https://docs.rs/once_cell/1.13.0/once_cell/sync/struct.OnceCell.html

Reply:
Yes, use use `once_cell::sync::Lazy`.